### DISCUSSION TITLE
Clarification on Vite prefix in "runtime" mode

### DISCUSSION THREAD

**Discussion by pastinepolenta** [runtime-env/import-meta-env#752] [Q&A]
I am using Vite+React and I have set env vars with the standard approach of prefixing them with `VITE_`

In order to try the 12-factor approach I have addedd import-meta-env and set `transformMode: "runtime"`

When I launch the build I get an error:

```
[import-meta-env] It seems you want to use a runtime environment variable prefixed by or equal to: `VITE_`, but this conflicts with Vite's environment variables.
```

While I kinda get the error (and I can solve it by setting `envPrefix: 'notused',` in the vite config, I did not expect this to happen and I don't see the "requirement" documented. I thought this was taken care automatically so that we can still use the prefix `VITE_` and switch config to "runtime" without further manipulation.

So my question is if:
- I am missing something/doing something wrong
- or if this is the expected behavior (meaning the "runtime" setting does not work out-the-box with `VITE_` prefix). 
- maybe some clarification in the documentation about this is needed?

Thanks

**Comment by soc221b** [comment]
It seems you putting the Vite's handled environment variable keys in your `.env.example`.

This won't work because import-meta-env use `.env.example` to determine which environment variable you want to use.

You can fix this by removing `VITE_` prefix-ed keys from `.env.example`.

**Reply by pastinepolenta** [reply]
Thanks I think I understand the fix but I fail to understand why this is a conflict in the first place. 
I would expect the plugin to simply replace the `VITE_` prefixed vars as it does for any other vars since that is the final goal anyways: to replace those vars with values coming from the globalThis.
Bear with me, maybe is just "as designed" and I will close this post. Hopefully you can clarify if my use-case makes sense or I am missing something.

As my codebase already uses `VITE_` prefixed vars (as it should), I don't want to rename those to a non-prefixed var. If in the `.example` file I use different variables, they are not substituted in the codebase cause obviously they don't exists.
Keeping the same variables as per Vite defaults, allows me to keep the same codebase and easily switch between a "buildonce" approach (with import-meta-env) and a "buildmany" approach (without the plugin) without impacting the variable naming scheme.

While this might be a specific requirement, I feel that all projects starting with Vite would end up like in the same boat: with a bunch of `VITE_` prefixed vars that we don't want to rename (not out of laziness but to keep the codebase as standard as possible)

The plugin is great and very flexible and takes into account multiple scenarios, so I wonder if I am missing something obvious here or if there something that can be solved or documented (as mentioned the `envPrefix: 'notused'` is a workaround)
